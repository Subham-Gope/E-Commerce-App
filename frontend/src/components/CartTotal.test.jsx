// import { describe, expect, test } from "vitest";
// import { screen, render } from "@testing-library/react";
// import ShopContext from "../context/ShopContext";
// import CartTotal from "./CartTotal";

// // Reusable render function
// const renderWithContext = (value) => {
//   return render(
//     <ShopContext.Provider value={value}>
//       <CartTotal />
//     </ShopContext.Provider>
//   );
// };

// // creating mock values
// const mockValues = {
//   currency: "$",
//   delivery_fee: 5,
//   getCartAmount: () => 120,
// };

// describe("Testing the CartTotal.jsx file", () => {
//   test("renders shipping fee, currency, total amount", () => {
//     // rendering with context
//     renderWithContext(mockValues);

//     // checking the title texts
//     const title = screen.getByText(/cart/i);
//     expect(title).toBeInTheDocument();

//     // checking the subtotal , total and currency
//     const subtotal = screen.getByText(/\$ 120.00/i);
//     expect(subtotal).toBeInTheDocument();
//     const shippingFee = screen.getByText(/\$ 5.00/i);
//     expect(shippingFee).toBeInTheDocument();
//     const total = screen.getByText(/\$ 125.00/i);
//     expect(total).toBeInTheDocument();
//   });
// });
