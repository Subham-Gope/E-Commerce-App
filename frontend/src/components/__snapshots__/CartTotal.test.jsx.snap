// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the CartTotal.jsx file > renders shipping fee, currency, total amount 1`] = `
<div>
  <div
    class="w-full"
  >
    <div
      class="text-2xl"
    >
      <div
        class="inline-flex gap-2 items-center mb-3"
      >
        <p
          class="text-gray-500"
        >
          CART
           
          <span
            class="text-gray-700 font-medium"
          >
            TOTAL
          </span>
        </p>
        <p
          class="w-8 sm:w-12 h-[1px] sm:h-[2px] bg-gray-700"
        />
      </div>
    </div>
    <div
      class="flex flex-col gap-2 mt-2 text-sm"
    >
      <div
        class="flex justify-between"
      >
        <p>
          Subtotal
        </p>
        <p>
          $ 120.00
        </p>
      </div>
      <hr />
      <div
        class="flex justify-between"
      >
        <p>
          Shipping Fee
        </p>
        <p>
          $ 5.00
        </p>
      </div>
      <hr />
      <div
        class="flex justify-between"
      >
        <b>
          Total
        </b>
        <b>
          $ 125.00
        </b>
      </div>
    </div>
  </div>
</div>
`;
